nrfcloud/apiClient.ts

```diff
@@ -1,71 +1,154 @@
+								]),
+							),
+						}),
+					),
+					device: Type.Optional(
+						Type.Object({
+							deviceInfo: Type.Optional(
+								Type.Partial(
+									Type.Object({
+										appVersion: Type.String(), // e.g. '1.1.0'
+										modemFirmware: Type.String(), // e.g. 'mfw_nrf9160_1.3.4'
+										imei: Type.String(), // e.g. '352656108602296'
+										board: Type.String(), // e.g. 'thingy91_nrf9160'
+										hwVer: Type.String(), // e.g. 'nRF9160 SICA B1A'
+									}),
+								),
+							),
+						}),
+					),
+				}),
+			),
+			desired: Type.Optional(
+				Type.Object({
+					config: Type.Optional(DeviceConfig),
+				}),
+			),
+			version: Type.Number(),
+		}),
+	),
+})
+
+const Page = <T extends TSchema>(Item: T) =>
+	Type.Object({
+		total: Type.Integer(),
+		items: Type.Array(Item),
+	})
+const Devices = Page(Device)
+
+const AccountInfo = Type.Object({
+	mqttEndpoint: Type.String(), // e.g. 'mqtt.nrfcloud.com'
+	mqttTopicPrefix: Type.String(), // e.g. 'prod/a0673464-e4e1-4b87-bffd-6941a012067b/',
+	team: Type.Object({
+		tenantId: Type.String(), // e.g. 'bbfe6b73-a46a-43ad-94bd-8e4b4a7847ce',
+		name: Type.String(), // e.g. 'hello.nrfcloud.com'
+	}),
+	plan: Type.Object({
+		currentMonthCosts: Type.Array(
+			Type.Object({
+				price: Type.Number(), // e.g. 0.1
+				quantity: Type.Number(), // e.g. 9
+				serviceDescription: Type.String(), // e.g. 'Devices in your account'
+				serviceId: Type.Union([
+					Type.Literal('Devices'),
+					Type.Literal('Messages'),
+					Type.Literal('SCELL'),
+					Type.Literal('MCELL'),
+				]),
+				total: Type.Number(), // e.g. 0.9
+			}),
+		),
+		currentMonthTotalCost: Type.Number(), // e.g. 2.73
+		name: Type.Union([Type.Literal('PRO'), Type.Literal('DEVELOPER')]),
+		proxyUsageDeclarations: Type.Object({
+			AGPS: Type.Number(), // e.g. 0
+			GROUND_FIX: Type.Number(), // e.g. 200
+			PGPS: Type.Number(), // e.g. 0
+		}),
+	}),
+	role: Type.Union([
+		Type.Literal('owner'),
+		Type.Literal('admin'),
+		Type.Literal('editor'),
+		Type.Literal('viewer'),
+	]),
+	tags: Type.Array(Type.String()),
+})
+
+const BulkOpsRequest = Type.Object({
+	bulkOpsRequestId: Type.String(), // e.g. '01EZZJVDQJPWT7V4FWNVDHNMM5'
+	endpoint: Type.Union([
+		Type.Literal('PROVISION_DEVICES'),
+		Type.Literal('REGISTER_PUBLIC_KEYS'),
+		Type.Literal('VERIFY_ATTESTATION_TOKENS'),
+		Type.Literal('VERIFY_JWTS'),
+		Type.Literal('CLAIM_DEVICE_OWNERSHIP'),
+	]), // e.g. 'PROVISION_DEVICES'
+	status: Type.Union([
+		Type.Literal('PENDING'),
+		Type.Literal('IN_PROGRESS'),
+		Type.Literal('FAILED'),
+		Type.Literal('SUCCEEDED'),
+	]), // e.g. 'PENDING'
+	requestedAt: Type.String({
+		pattern:
+			'^\\d{4}-[01]\\d-[0-3]\\dT[0-2]\\d:[0-5]\\d:[0-5]\\d.\\d+([+-][0-2]\\d:[0-5]\\d|Z)$',
+	}), // e.g. '2020-06-25T21:05:12.830Z'
+	completedAt: Type.Optional(
+		Type.String({
+			pattern:
+				'^\\d{4}-[01]\\d-[0-3]\\dT[0-2]\\d:[0-5]\\d:[0-5]\\d.\\d+([+-][0-2]\\d:[0-5]\\d|Z)$',
+		}),
```

Comment:
Turn into a re-usable type.
```suggestion
		DateString,
```

```typescript
const DateString = Type.String({
			pattern:
				'^\\d{4}-[01]\\d-[0-3]\\dT[0-2]\\d:[0-5]\\d:[0-5]\\d.\\d+([+-][0-2]\\d:[0-5]\\d|Z)$',
title: 'ISO Date string'
		})
```